frontend_app_file/src/container/PopupCreateFile.jsx

```diff
@@ -137,39 +186,81 @@ class PopupCreateFile extends React.Component {
+        const uploadedFiles = state.uploadedFiles
         switch (xhr.status) {
           case 200:
             const jsonResult200 = JSON.parse(xhr.responseText)
-            this.handleClose()
-
-            GLOBAL_dispatchEvent({ type: CUSTOM_EVENT.REFRESH_CONTENT_LIST, data: {} })
-
-            GLOBAL_dispatchEvent({
-              type: CUSTOM_EVENT.OPEN_CONTENT_URL,
-              data: {
-                workspaceId: jsonResult200.workspace_id,
-                contentType: state.appName,
-                contentId: jsonResult200.content_id
-              }
-            })
+
+            filePosted.jsonResult = { ...jsonResult200, code: 200 }
+
+            uploadedFiles.push(filePosted)
+            this.setState({ uploadedFiles })
             break
           case 400:
             const jsonResult400 = JSON.parse(xhr.responseText)
+
+            let hasError = props.t('Error while creating file')
             switch (jsonResult400.code) {
-              case 3002: this.sendGlobalFlashMessage(props.t('A content with the same name already exists')); break
-              case 6002: this.sendGlobalFlashMessage(props.t('The file is larger than the maximum file size allowed')); break
-              case 6003: this.sendGlobalFlashMessage(props.t('Error, the shared space exceed its maximum size')); break
-              case 6004: this.sendGlobalFlashMessage(props.t('You have reach your storage limit, you cannot add new files')); break
+              case 3002: hasError = props.t('A content with the same name already exists'); break
+              case 6002: hasError = props.t('The file is larger than the maximum file size allowed'); break
+              case 6003: hasError = props.t('Error, the shared space exceed its maximum size'); break
+              case 6004: hasError = props.t('You have reach your storage limit, you cannot add new files'); break
             }
+            filePosted.jsonResult = jsonResult400
+            filePosted.hasError = hasError
+            uploadedFiles.push(filePosted)
+            this.setState({ uploadedFiles })
+            break
+          default:
+            filePosted.jsonResult = { code: 0 }
+            filePosted.hasError = props.t('Error while creating file')
+            uploadedFiles.push(filePosted)
+            this.setState({ uploadedFiles, progressUpload: { display: false } })
             break
-          default: this.sendGlobalFlashMessage(props.t('Error while creating file')); break
         }
       }
     }
 
     xhr.send(formData)
   }
 
+  handleValidate = async () => {
+    const { state } = this
+
+    this.setState({
+      progressUpload: {
+        percent: 0
+      },
+      uploadStarted: true
+    })
+
+    state.uploadFiles.forEach((file, index) => this.postFile(file, index))
+  }
+
+  handleDeleteFile = (file) => {
+    const { state } = this
+
+    const uploadFiles = state.uploadFiles
+    uploadFiles.splice(uploadFiles.indexOf(file), 1)
```

Comment:
avoid using splice since this function mutate the source array. Prefer immutable functions